hazelcast-client-legacy/src/test/java/com/hazelcast/client/cache/nearcache/ClientCacheSerializationCountTest.java

```diff
@@ -0,0 +1,266 @@
+
+    static class SerializationCountingData implements Portable {
+        public static int FACTORY_ID = 1;
+        public static int CLASS_ID = 1;
+
+        public SerializationCountingData() {
+
+        }
+
+        @Override
+        public int getFactoryId() {
+            return FACTORY_ID;
+        }
+
+        @Override
+        public int getClassId() {
+            return CLASS_ID;
+        }
+
+        @Override
+        public void writePortable(PortableWriter writer) throws IOException {
+            serializeCount.incrementAndGet();
+        }
+
+        @Override
+        public void readPortable(PortableReader reader) throws IOException {
+            deserializeCount.incrementAndGet();
+        }
+
+    }
+
+    @After
+    public void tearDown() {
+        hazelcastFactory.terminateAll();
+    }
+
+    protected CacheConfig createCacheConfig(InMemoryFormat inMemoryFormat) {
+        return new CacheConfig()
+                .setName(CACHE_NAME)
+                .setInMemoryFormat(inMemoryFormat);
+    }
+
+    protected NearCacheConfig createNearCacheConfig(InMemoryFormat inMemoryFormat
+            , NearCacheConfig.LocalUpdatePolicy localUpdatePolicy) {
+        return new NearCacheConfig()
+                .setName(CACHE_NAME)
+                .setLocalUpdatePolicy(localUpdatePolicy)
+                .setInMemoryFormat(inMemoryFormat);
+    }
+
+    protected Config createConfig() {
+        Config config = new Config();
+        SerializationConfig serializationConfig = config.getSerializationConfig();
+        prepareSerializationConfig(serializationConfig);
+        return config;
+    }
+
+    protected ClientConfig createClientConfig() {
+        ClientConfig config = new ClientConfig();
+        SerializationConfig serializationConfig = config.getSerializationConfig();
+        prepareSerializationConfig(serializationConfig);
+        return config;
+    }
+
+    private void prepareSerializationConfig(SerializationConfig serializationConfig) {
+        ClassDefinition classDefinition = new ClassDefinitionBuilder(SerializationCountingData.FACTORY_ID,
+                SerializationCountingData.CLASS_ID).build();
+        serializationConfig.addClassDefinition(classDefinition);
+
+        serializationConfig.addPortableFactory(SerializationCountingData.FACTORY_ID, new PortableFactory() {
+            @Override
+            public Portable create(int classId) {
+                return new SerializationCountingData();
+            }
+        });
+    }
+
+    @Test
+    public void testDeserializationCountWith_ObjectNearCache_cacheLocalUpdatePolicy() {
+        NearCacheConfig nearCacheConfig = createNearCacheConfig(InMemoryFormat.OBJECT
+                , NearCacheConfig.LocalUpdatePolicy.CACHE);
+        prepareCache(nearCacheConfig);
+
+        final String key = randomString();
+        SerializationCountingData value = new SerializationCountingData();
+        cache.put(key, value);
+        assertAndReset(1, 0);
```

Comment:
since putting to near-cache is done as async, it would be better making this assertion after `assertEquals(1, nearCache.size());` so we can be sure that deserialization count is `0` after the record is put to near-cache